google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1284,4 +1305,29 @@ public void deleteNestedFieldUsingFieldPath() throws Exception {
     documentSnapshots = documentReference.get().get();
     assertNull(documentSnapshots.getData().get("c.d"));
   }
+
+  /** Wrapper around ApiStreamObserver that returns the results in a list. */
+  private static class StreamConsumer<T> implements ApiStreamObserver<T> {
+    SettableApiFuture<List<T>> done = SettableApiFuture.create();
+    List<T> results = new ArrayList<>();
```

Comment:
```suggestion
    List<T> results = Collections.synchronizedList(new ArrayList<>());
```

Very annoying subtle to debug if there happens to be no other synchronization between the writes and reads of this list.